assembl/static2/js/app/components/common/nuggetsManager.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+import Nuggets from '../debate/thread/nuggets';
+
+const invalidIndex = (index) => {
+  return index === -1;
+};
+
+class NuggetsManager {
+  nuggetsList: Array<Nuggets>;
+  constructor() {
+    this.nuggetsList = [];
+  }
+  add(nuggets: Nuggets) {
+    this.nuggetsList.push(nuggets);
+    this.sort();
+    this.update();
+  }
+  update = () => {
+    let previous = null;
```

Comment:
No no. Please don't use mutations like this. You're using a functional approach, keep to it.
```
this.nuggestList.forEach( (nugget, index, nuggets) => {
  if (index == 0) {
    nugget.updateTop(null);
  }
  nugget.updateTop(nuggets[index-1]);
});